grafana-plugin/src/network/index.ts

```diff
@@ -34,14 +38,63 @@ export const makeRequest = async <RT = any>(path: string, config: RequestConfig)
   const { method = 'GET', params, data, validateStatus } = config;
 
   const url = `${API_PROXY_PREFIX}${API_PATH_PREFIX}${path}`;
+  const otel = FaroHelper.faro?.api?.getOTEL();
+
+  if (FaroHelper.faro && otel) {
+    const tracer = otel.trace.getTracer('default');
+    let span = otel.trace.getActiveSpan();
+
+    if (!span) {
+      span = tracer.startSpan('http-request');
+      span.setAttribute('page_url', document.URL.split('//')[1]);
+      span.setAttribute(SemanticAttributes.HTTP_URL, url);
+      span.setAttribute(SemanticAttributes.HTTP_METHOD, method);
+    }
+
+    return new Promise<RT>((resolve, reject) => {
```

Comment:
would it be possible to de-duplicate this a bit? The two `try/catch` clauses are nearly identical and there's quite a lot of duplication. Also, any reason why to mix and match `async/await` and `Promise` in the same function?

Reply:
I'll change it so that it's using just async/await, but I will leave it as 2 separate blocks for now because the first one relies on OTEL implementation and actually calls methods from OTEL, while the other one is just a try catch with log. It will also benefit to have them separate in case we want to add more span attributes on the open telemetry side.